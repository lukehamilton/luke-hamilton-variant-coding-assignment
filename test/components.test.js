import React from 'react'
import renderer from 'react-test-renderer'
import { shallow, mount, configure } from 'enzyme'

import Adapter from 'enzyme-adapter-react-16'

configure({ adapter: new Adapter() })

import Header from '../components/header'
import Item from '../components/item'
import SuggestionInput, { StyledButton } from '../components/suggestion-input'

import sampleVariants from './sample-data/variants'

describe('Header', () => {
  test('it renders correctly', () => {
    expect(renderer.create(<Header />).toJSON()).toMatchSnapshot()
  })
})

describe('SuggestionInput', () => {
  test('it renders correctly', () => {
    expect(renderer.create(<SuggestionInput />).toJSON()).toMatchSnapshot()
  })
  test('it triggers handleOnSubmit on submit button click', () => {
    const handleOnSubmit = jest.fn()
    const el = shallow(<SuggestionInput handleOnSubmit={handleOnSubmit} />, {
      disableLifecycleMethods: true
    })
    el.find(StyledButton).simulate('click')
    const calls = handleOnSubmit.mock.calls
    expect(calls.length).toBe(1)
  })
  test('it triggers handleOnSubmit on submit button click', () => {
    const handleOnSubmit = jest.fn()
    const el = mount(<SuggestionInput handleOnSubmit={handleOnSubmit} />, {
      disableLifecycleMethods: true
    })
    el.find(StyledButton).simulate('click')
    const calls = handleOnSubmit.mock.calls
    expect(calls.length).toBe(1)
  })
})

describe('Table Item Row', () => {
  test('it renders correctly shaded', () => {
    expect(
      renderer.create(<Item data={sampleVariants[0]} shaded={true} />).toJSON()
    ).toMatchSnapshot()
  })
  test('it renders correctly not shaded', () => {
    expect(
      renderer.create(<Item data={sampleVariants[1]} shaded={false} />).toJSON()
    ).toMatchSnapshot()
  })
})
