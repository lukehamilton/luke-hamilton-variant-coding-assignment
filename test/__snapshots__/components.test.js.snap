// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header it renders correctly 1`] = `
<div
  className="css-1yryf18-Container e10so7i70"
>
  <div
    className="css-1gvxlg9-Container e14bnm7i0"
  >
    <div
      className="css-ss2xpo-InputWrapper e14bnm7i2"
    >
      <input
        className="css-ox8hf7-Input e1i9eloy0"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search for a gene"
      />
    </div>
    <button
      className="css-1x0t62n-StyledButton e14bnm7i1"
      onClick={[Function]}
    >
      Submit
    </button>
  </div>
  <div
    className="css-ylx3i7-TableHead e10so7i72"
  >
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Gene
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Nucleotide Change
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Protein Change
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Alias
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Region
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Reported Classification
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Last Evaluated
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      Last Updated
    </div>
    <div
      className="css-qbgrms-HeadField e10so7i71"
    >
      More Info
    </div>
  </div>
</div>
`;

exports[`SuggestionInput it renders correctly 1`] = `
<div
  className="css-1gvxlg9-Container e14bnm7i0"
>
  <div
    className="css-ss2xpo-InputWrapper e14bnm7i2"
  >
    <input
      className="css-ox8hf7-Input e1i9eloy0"
      onChange={[Function]}
      onKeyDown={[Function]}
      placeholder="Search for a gene"
    />
  </div>
  <button
    className="css-1x0t62n-StyledButton e14bnm7i1"
    onClick={[Function]}
  >
    Submit
  </button>
</div>
`;

exports[`Table Item Row it renders correctly not shaded 1`] = `
<div
  className="css-qd2p6w-Container eb9w5ky1"
>
  <div
    className="css-19zq68w-Row eb9w5ky3"
    onClick={[Function]}
  >
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      AMY1A
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      not provided
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      2017-09-14
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      <a
        href="https://www.ncbi.nlm.nih.gov/clinvar/RCV000161176"
        target="_blank"
      >
        ClinVar
      </a>
    </div>
  </div>
</div>
`;

exports[`Table Item Row it renders correctly shaded 1`] = `
<div
  className="css-vq9p01-Container eb9w5ky1"
>
  <div
    className="css-19zq68w-Row eb9w5ky3"
    onClick={[Function]}
  >
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      RHD
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      not provided
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      2017-09-14
    </div>
    <div
      className="css-vqbiuz-Field eb9w5ky2"
    >
      <a
        href="https://www.ncbi.nlm.nih.gov/clinvar/RCV000161157"
        target="_blank"
      >
        ClinVar
      </a>
    </div>
  </div>
</div>
`;
